x/group/keeper/msg_server.go

```diff
@@ -712,16 +685,25 @@ func doTally(ctx sdk.Context, p *group.Proposal, electorate group.GroupInfo, pol
 	if err != nil {
 		return err
 	}
-	switch result, err := policy.Allow(p.FinalTallyResult, electorate.TotalWeight, ctx.BlockTime().Sub(submittedAt)); {
+
+	tallyResult, err := k.Tally(ctx, *p, policyInfo.GroupId)
+	if err != nil {
+		return err
+	}
+
+	switch result, err := policy.Allow(tallyResult, electorate.TotalWeight, ctx.BlockTime().Sub(submittedAt)); {
```

Comment:
It seems a little redundant now to to have the `sinceSubmission` argument. I assume that was also used to enforce a voting period.

Reply:
Why is it redundant? The `sinceSubmission` argument is used in the `Allow` implementation to check against the decision policy's MinExecutionPeriod.

Reply:
Yeah sorry I didn't realise where the check for the min execution period was being made. This is good now